src/__tests__/actions/testLoginAction.js

```diff
@@ -0,0 +1,46 @@
+import moxios from "moxios";
+import thunk from "redux-thunk";
+import configureMockStore from "redux-mock-store";
+import * as types from "../../redux/action_types";
+import { loginError, loginSuccessful } from "../../redux/actions/loginAction";
+
+const middleware = [thunk];
+const mockstore = configureMockStore(middleware);
+const emailError = "email is required";
+const loginSuccefullMessage = "mike";
+const regURL = "http://localhost:8000/api/users/login";
+
+describe("action", () => {
+  beforeEach(() => moxios.install());
+  afterEach(() => moxios.uninstall());
+
+  it("should dispatch login successful", async done => {
```

Comment:
`done` is a representation of `()` in this case. Replace with `()` and delete line 26. Please do the same for all scenarios you've used done to avoid adding unnecessary extra lines of code.